.github/workflows/test.yml

```diff
@@ -9,9 +9,58 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Determine Go version from go.mod
+        run: echo "GO_VERSION=$(grep "go 1." go.mod | cut -d " " -f 2)" >> $GITHUB_ENV
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
       - name: Lint
         run: make lint
       - name: Test
         run: make test
+  e2e-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - kind-node-version: v1.18.8
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine Go version from go.mod
+        run: echo "GO_VERSION=$(grep "go 1." go.mod | cut -d " " -f 2)" >> $GITHUB_ENV
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Run e2e tests
+        run: make install_bats setup_e2e_test e2e_test -e KIND_NODE_VERSION=${{ matrix.kind-node-version }} -e KIND_KUBECTL_ARGS=--validate=false
+      - name: Show e2e debug logs
+        run: cat e2e/debug/detik/*
+  image:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
Doesn't really make sense to setup go here, since the build is happening inside the Dockerfile anyway, right?

Reply:
`make docker` depends  on `$(BINARY_NAME)` which uses Go to build the binary.. which it then additionally builds within docker.  
I'm not sure about that part, also mostly copied it. I'll probably remove it again and move docker image build to after test run, as the way it is now, it's run in parallel and run even if tests fail.

Agree?